x-pack/plugins/cloud/public/types.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+
+export interface CloudStart {
+  /**
+   * A React component that provides a pre-wired `React.Context` which connects components to Cloud services.
+   */
+  CloudContextProvider: FC<{}>;
```

Comment:
Took this opportunity to move the browser-side contracts to a dedicated file, given they started to become quite big.